server/core/config/valid/global_cfg.go

```diff
@@ -521,3 +527,16 @@ func (g GlobalCfg) MatchingRepo(repoID string) *Repo {
 	}
 	return nil
 }
+
+// RepoConfigFile returns a repository specific file path
+// If not defined, return atlantis.yaml as default
+func (g GlobalCfg) RepoConfigFile(repoID, branchName string) string {
+	for _, repo := range g.Repos {
+		if repo.IDMatches(repoID) &&
+			repo.BranchMatches(branchName) &&
+			repo.RepoConfigFile != "" {
+			return repo.RepoConfigFile
+		}
+	}
+	return DefaultAtlantisFile
+}
```

Comment:
I implemented this with `repoId` and `branchName` filters. By adding `branchName` filter, we can use multiple atlantis servers when using different base branches.